### DISCUSSION TITLE
links with a path

### DISCUSSION THREAD

**Discussion by dandobos** [SilentVoid13/Templater#738] [Help]
I love templator but I have a problem.

In order to move a file to a folder based on template chosen, I have been using: <% await tp.file.move("/Example Folder/" + tp.file.title) %> 

But that causes a problem. It means that when I link to that file it will put the path in the link. 

So it's: [[Example Folder/Note Topic]]
Instead of what I want which is just: [[Note Topic]]

When I remove  <% await tp.file.move("/Example Folder/" + tp.file.title) %> 
there's no problem.

I'd love to use templator to automatically move the note to a folder, but this issue is a problem.

I'm grateful for your help.

**Comment by welpdx** [comment]
I can't replicate the error. Is it perhaps a due to duplicates. When a file has a duplicate name than another file, the internal links is updated to [include paths in the internal link](https://i.imgur.com/ISHnXyP.jpeg). You will either see a [pop up dialogue](https://i.imgur.com/2RwYEkp.jpeg) or a Notice() telling you an internal link has been updated. 

Try to make your new files unique in that template so that each file generated is new. 

---

If this isnt your issue, can you include more information? (i.e., your templater, obs version, a demo, etc) as this will help to pinpoint problem. Thanks!

**Reply by dandobos** [reply]
Thank you for your help.

I tried to duplicate the problem but failed.  Then I noticed that I was creating duplicates somehow so perhaps that is what caused it.

The way I seem to have created a duplicate is
1. Create new note
2. Apply template with templater
3. Attempt to change the name of the note... which doesn't change the name, it only seems to create the duplicate

The template that I'm using is:
---
Summary:

---
<% await tp.file.move("/Permanent Notes/" + tp.file.title) %>
# <% tp.file.title %>



## Meta info
Concepts:
Topics:

Created: <% tp.date.now("ddd Do MMM, YYYY HH:mm") %>
Modified: <%+ tp.file.last_modified_date("ddd Do MMM, YYYY  HH:mm") %>
Note type: #pn 
FollowupThen status: #fup/todo



## Reference

## See also




I'm using 0.15.9 Obsidian

Something super strange is that dataview detects two notes which seem to be identical, but I can only locate one note.
https://share.getcloudapp.com/7Kukvzbo

**Reply by dandobos** [reply]
Now my issue is that I need to change the name of the note as well.
The error I get when I try to change the name is "Destination file already exists"

**Reply by dandobos** [reply]
I quit and re-opened obsidian and duplicate is gone. Super odd.

**Reply by welpdx** [reply]
That is quite strange. 

#### Clarification questions
1. I am guessing in step `1. Create new note`, you change the title from "Untitled" to something unique in your vault right? 
2. It is after this that you then run the `Templater: Open Insert Template Modal` for step 2 right?
3. What do you mean in step 3 that the name of the note doesn't change? Isn't it supposed to stay the same but with only the path being changed?

**Reply by dandobos** [reply]
1. Yes
2. Yes
3. This was a bug that I can't reproduce. But at the time, I wanted to change the title and it didn't change. It gave error "Destination file already exists". But then I quit and re-opened Obsidian and everything was fine ie I could change note title.

Thank you for your help. I'm not sure why it happened, but it works fine now.

**Reply by welpdx** [reply]
Yea I've seen that "Destination file already exists" quite a bit. Causes are different, but its usually due to obsidian thinking that a particular file or folder already exists. e.g., #318 #494

But its good to hear that its working now. If something comes up, feel free to note how you created the error here.

**Comment by dandobos** [comment]
Hello again, the bug is back :(

In settings, I have:  Files & Links > New Link Format set to "Shortest path possible"
![image](https://user-images.githubusercontent.com/1361719/188514695-7b8655af-0a3c-4308-9a07-b5065cc76f4c.png)

When I create a note using this template, it always puts the path with "Permanent Path" ie
[[Permanent notes/test1d (PN)]]
instead of just
[[test1d (PN)]]

Note that I am using a base note to choose the template. So my process is:
1. Open file
2. Type: test1d (PN)
3. Press: Return
4. File is created

I am most grateful for your help

**Comment by dandobos** [comment]
Also, I only seem to have the problem with PN template but not LN template.
Here are the files
[LN generic (template).md](https://github.com/SilentVoid13/Templater/files/9492299/LN.generic.template.md)
[Base Note (template).md](https://github.com/SilentVoid13/Templater/files/9492298/Base.Note.template.md)
[PN (template).md](https://github.com/SilentVoid13/Templater/files/9492297/PN.template.md)

**Comment by dandobos** [comment]
@welpdx - any thoughts?

**Reply by welpdx** [reply]
lemme take a look
